### DISCUSSION TITLE
Chapter 2 challenge issue

### DISCUSSION THREAD

**Discussion by infinite969** [Cyfrin/foundry-full-course-cu#693] [General]
Hey friends, I am confused on how to approach challenge 2. Still new to solidity so bear with me haha. I copied the first bit of code in part 1 of 6  from Arbiscan and I am not able to import: {AFoundryCourseChallenge} from "AFoundryCourseChallenge.sol"; 

because:
Error: not found AFoundryCourseChallenge.sol

I think this is because "AFoundryCourseChallenge.sol"; is not saved in my IDE, but honestly I have no idea if I am even approaching the challenge correctly. If someone can let me know how to solve this issue, or can point me in the right direction as to what I am supposed to be doing, that would be greatly appreciated.

Here is the code 1 of 6 from Arbiscan, I added my twitter handle in the "solveChallenge"  function because I figured that is what I am supposed to do.

```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {AFoundryCourseChallenge} from "AFoundryCourseChallenge.sol";

contract LessonOne is AFoundryCourseChallenge {
    string private constant LESSON_IMAGE = "ipfs://QmYNk9TaRp9QDRjDZYxrwh5xQGMeC9zY23tB4fs5qby6KT";

    constructor(address fcn) AFoundryCourseChallenge(fcn) {}

    /*
     * CALL THIS FUNCTION!
     * 
     * @param yourTwitterHandle - Your twitter handle. Can be a blank string.
     */
    function solveChallenge(string memory infinite96969) external {
        _updateAndRewardSolver(infinite96969);
    }

    ////////////////////////////////////////////////////////////////////////////////////////////////////////////
    /////////////////// The following are functions needed for the NFT, feel free to ignore. ///////////////////
    ////////////////////////////////////////////////////////////////////////////////////////////////////////////
    function description() external pure override returns (string memory) {
        return "Cyfrin Foundry Full Course: Awesome! You got the basics down! You minted an NFT!";
    }

    function attribute() external pure override returns (string memory) {
        return "Getting learned!";
    }

    function specialImage() external pure override returns (string memory) {
        return LESSON_IMAGE;
    }
}
```

thanks lads

**Comment by devmousa** [comment]
you don't have to change anything in the code
just go to ```write contract``` section in the arbiscan and connect your wallet, then click on solveChallenge() function, fill the parameter with your own twitter handle, then confirm the transaction
and if you don't have arbitrum currency (I don't know a lot about it), just use sepolia.etherscan (with sepoliaETH testnet money)

I hope you understand me, if you don't understand, feel free to tell me and I'll try attach some images to explain more
happy coding ;)

**Reply by Olyray** [reply]
Aren't we supposed to solve the challenge in the Remix IDE?

**Comment by infinite969** [comment]
oh great! thank you. I connected my wallet and when I clicked the 'write'  in  renounceOnwership I got this notification in my metamask:

"We were not able to estimate gas. There might be an error in the contract and this transaction may fail." 
It is giving me the option to proceed anyway but the transaction failed when I tried just now. Anything that I can do to solve this issue?

**Reply by devmousa** [reply]
you have to call ```solveChallenge``` and don't call any function except it
and in the ```solveChallenge|```, fill the parameter with your twitter handler and send the transaction and congratulations, you've solved the challenge
(when you're going through the course you'll get it more and it'll make sense more to you)

try the solution I mentioned above and let me know if it worked or not
(*Note:* you have just to call the ```solveChallenge``` function, then you'll find the NFT is sent to your waller, then you'll have just to import it in your metamask)

**Reply by ahpoh1349** [reply]
hi Ser but now it says input value above it cannot be empty for the number(uint)
![challenge input value cannot be empty](https://github.com/user-attachments/assets/8472cfec-9ec9-43ba-b996-e33a82310c47)

**Reply by EngrPips** [reply]
Yeah, you need to provide value for the function argument.

**Comment by infinite969** [comment]
ah okay it looks like I got the nft 
tx hash: 0x3467e720619d87fc04e56a8488a6040e73bff4b87c6129bdbbfb1604665715d6
its not showing up in my wallet, I will find where to import it

**Reply by devmousa** [reply]
you can import it from nft section in your metamask

**Reply by infinite969** [reply]
Got it. you're the best man, thank you for your help!

**Reply by devmousa** [reply]
you're welcome, any time ❤️
